src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageModelActivatorProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
+{
+    /// <summary>
+    /// <see cref="IPageActivatorProvider"/> that uses type activation to create Pages.
+    /// </summary>
+    public class DefaultPageModelActivatorProvider : IPageModelActivatorProvider
+    {
+        private readonly Action<PageContext, object> _disposer = Dispose;
+
+        /// <inheritdoc />
+        public virtual Func<PageContext, object> CreateActivator(CompiledPageActionDescriptor actionDescriptor)
+        {
+            if (actionDescriptor == null)
+            {
+                throw new ArgumentNullException(nameof(actionDescriptor));
+            }
+
+            var modelTypeInfo = actionDescriptor.ModelTypeInfo?.AsType();
+            if (modelTypeInfo == null)
+            {
+                throw new ArgumentException(Resources.FormatPropertyOfTypeCannotBeNull(
+                    nameof(actionDescriptor.ModelTypeInfo),
+                    nameof(actionDescriptor)),
+                    nameof(actionDescriptor));
+            }
+
+            var factory = ActivatorUtilities.CreateFactory(modelTypeInfo, Type.EmptyTypes);
+            return (context) => factory(context.HttpContext.RequestServices, EmptyArray<object>.Instance);
```

Comment:
It would be dope to put this part into a virtual method, then there's less boilerplate if you want to use something other than `ActivatorUtilities`.